### PR TITLE
Deprecate `hesa_itt_data` vendor API field

### PR DESCRIPTION
**Pull request by elliotcm**
The HESA code fields (sex/disability/ethnicity) will return `nil` or `[]` for applications in the 2023 cycle.

Note: This **does not** change the API used by Register.

## Context

These fields are being replaced to account for longer HESA codes.

#### Before
Screenshot 2022-10-07 at 15 42 24
Screenshot 2022-10-07 at 15 42 11

#### After
Screenshot 2022-10-07 at 15 41 45
Screenshot 2022-10-07 at 15 41 32

##### Before
```
            "hesa_itt_data": {
                "sex": "10",
                "disability": [
                    "96"
                ],
                "ethnicity": "31",
                "other_disability_details": "Acquired brain injury",
                "other_ethnicity_details": null
            },
```
##### After
```
            "hesa_itt_data": {
                "disability": [
                    "96"
                ],
                "ethnicity": "31",
                "sex": "10",
                "other_disability_details": "Acquired brain injury",
                "other_ethnicity_details": null
            },
```
##### Before
```
            "hesa_itt_data": {
                "sex": "12",
                "disability": [
                    "51"
                ],
                "ethnicity": "50",
                "other_disability_details": null,
                "other_ethnicity_details": null
            },
```
##### After
```
            "hesa_itt_data": {
                "disability": [],
                "ethnicity": null,
                "sex": null,
                "other_disability_details": null,
                "other_ethnicity_details": null
            },
```

## Link to Trello card

https://trello.com/c/tvrmz5D4/748-vendor-api-null-out-sex-ethnicity-disability-hesa-codes-for-the-new-cycle

## Things to check

- [x] API release notes have been updated if necessary

### CODE REVIEW

**Comment by elliotcm** [comment]
[Vendor API: null out sex/ethnicity/disability HESA codes for the new cycle](https://trello.com/c/tvrmz5D4/748-vendor-api-null-out-sex-ethnicity-disability-hesa-codes-for-the-new-cycle)

File: app/presenters/concerns/hesa_itt_data_api_data.rb [hunk 1]
```diff
@@ -35,4 +29,20 @@ def other_ethnicity_details(equality_and_diversity_data)
 
     equality_and_diversity_data['ethnic_background']
   end
+
+  def hesa_codes(equality_and_diversity_data)
+    if application_form.recruitment_cycle_year >= 2023
```
Comment (carlosmartinez) [inline app/presenters/concerns/hesa_itt_data_api_data.rb:34]: would be nice to extract `2023` as a constant, eg `HESA_CODES_UPDATE_YEAR`

**Review by carlosmartinez — APPROVED** [review]

File: app/presenters/concerns/hesa_itt_data_api_data.rb (hunk 1 above)
Comment (elliotcm) [reply app/presenters/concerns/hesa_itt_data_api_data.rb]: Done.

**Comment by elliotcm** [comment]
> ## You have one or more flakey tests on this branch! ❄️ ❄️ ❄️
> Failed 1 out of 3 times at [./spec/workers/detect_invariants_daily_check_spec.rb:173](/DFE-Digital/apply-for-teacher-training/blob/748-nullify-hesa-codes-in-api/./spec/workers/detect_invariants_daily_check_spec.rb#L173): ⚠️ Sentry received :capture_exception with unexpected arguments expected: (#<DetectInvariantsDailyCheck::ObsoleteFeatureFlags:0x00007ff712152d48>) got: (#<DetectInvariantsDailyCheck::ObsoleteFeatureFlags:0x00007ff712153c98>) Diff: @@ -1 +1 @@ -[#<DetectInvariantsDailyCheck::ObsoleteFeatureFlags: The following obsolete feature flags have yet to be deleted from the database: ducimus_beatae_assumenda, ea_voluptatum_qui, eaque_iste_sunt, laudantium_placeat_odio, and mollitia_cupiditate_ad>] +[#<DetectInvariantsDailyCheck::ObsoleteFeatureFlags: The following obsolete feature flags have yet to be deleted from the database: ducimus_beatae_assumenda, eaque_iste_sunt, ea_voluptatum_qui, laudantium_placeat_odio, and mollitia_cupiditate_ad>]

I swear I fixed this already.

File: app/views/api_docs/vendor_api_docs/pages/release_notes.md [hunk 2]
```diff
@@ -1,3 +1,10 @@
+## 7th October 2022
+
+Documentation:
+
+Mark `hesa_itt_data` as deprecated. HESA code fields (sex/disability/ethnicity) within this object will contain `nil` or an empty array
+for applications in the 2023 recruitment cycle.
```
Comment (frankieroberto) [inline app/views/api_docs/vendor_api_docs/pages/release_notes.md:6]:
```suggestion
for applications in the 2022 to 2023 recruitment cycle.
```

**Review by frankieroberto — APPROVED** [review]

**Pull request merged**